### PR TITLE
Feature/serial settings

### PR DESCRIPTION
**Pull request by ryan-summers**
This is a proof-of-concept using the `serial-settings` crate to see how well it works with the API.

cc @jordens

### CODE REVIEW

**Comment by ryan-summers** [comment]
@jordens I had to refactor to the latest versions in Stabilizer. Mind taking a second look here?

File: src/hardware/serial_terminal.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            .map_err(Self::Error::Flash)?;
+        Ok(())
+    }
+
+    /// Execute a platform specific command.
+    fn cmd(&mut self, cmd: &str) {
+        match cmd {
+            "reboot" => {
+                cortex_m::interrupt::disable();
+                platform::shutdown_channels();
```
Comment (jordens) [inline src/hardware/serial_terminal.rs:52]: Where's the reboot?

**Review by jordens — COMMENTED** [review]
I'm confused. Don't we store the main board settings (broker etc) already in the eeprom? And with this now also in flash?
Could you document somewhere the overall architecture (what's stored where in which format, what's the precedence etc)?

**Comment by ryan-summers** [comment]
> I'm confused. Don't we store the main board settings (broker etc) already in the eeprom?

Correct, we did store it in EEPROM, but the EEPROM was too short to contain i.e. a DNS-named broker, so this PR is migrating existing settings over to flash.

> And with this now also in flash?

The intent of the design is to:
1. Load initial settings from EEPROM (if present)
2. Update settings from any flash-based values

That way, we don't cause users to lose their existing configs when switching from earlier booster firmware to latest. However, it does result in us carrying over a bunch of legacy EEPROM code for settings management and makes the whole process more complicated.

We _could_ just entirely deprecate the old EEPROM settings and force users to reconfigure booster during the firmware update if we want to simplify.

>  Could you document somewhere the overall architecture (what's stored where in which format, what's the precedence etc)?

Yeah, this is a good idea. I'll add it to one of the settings modules.

File: src/hardware/serial_terminal.rs (hunk 1 above)
Comment (ryan-summers) [reply src/hardware/serial_terminal.rs:52]: Woops! Indeed it was missing. I fixed it in f06e361677b0c133969694f001589ca9978d154b

File: src/settings/global_settings.rs [hunk 2]
```diff
@@ -153,16 +228,30 @@ impl BoosterMainBoardData {
...
+    pub fn reload(&mut self, storage: &mut Flash) {
+        let mut buffer = [0u8; 512];
+        storage.read(0, &mut buffer).unwrap();
+        let Ok(mut settings) = postcard::from_bytes::<Self>(&buffer) else {
+            return;
+        };
+
+        settings.mac = self.mac;
+        settings.version = self.version;
+        *self = settings;
```
Comment (jordens) [inline src/settings/global_settings.rs:252]:
```suggestion
        *self = settings;
        log::info!("Loaded settings from Flash");
```

File: src/settings/global_settings.rs [hunk 3]
```diff
@@ -197,7 +286,7 @@ impl BoosterMainBoardData {
             return Err(Error::Invalid);
         }
 
-        Ok((config, modified))
+        Ok((config.with_mac(eui48), modified))
```
Comment (jordens) [inline src/settings/global_settings.rs:289]:
```suggestion
        log::info!("Loaded settings from EEPROM");
        Ok((config.with_mac(eui48), modified))
```

File: src/settings/global_settings.rs [hunk 4]
```diff
@@ -1,14 +1,34 @@
 //! Booster NGFW NVM settings
-
-use crate::{hardware::Eeprom, Error};
+//!
+//! # Design
+//! Booster firmware maintained settings in an EEPROM on the main board until after v0.5.0. After
+//! v0.5.0, all global firmware settings for the device were moved to internal flash storage.
```
Comment (jordens) [inline src/settings/global_settings.rs:5]:
```suggestion
//! Booster firmware maintains mainboard settings in an EEPROM on the main board until v0.5.0. After
//! v0.5.0, main board settings are loaded from and stored to internal flash storage.
```

File: src/settings/global_settings.rs [hunk 5]
```diff
@@ -1,14 +1,34 @@
...
+//! from v0.5.0 firmware and earlier, the settings loading process occurs as follows:
+//! 1. Settings are loaded from Booster mainboard EEPROM
+//! 2. Settings are then loaded from flash (if possible)
+//!
+//! Any further saves to settings are persisted to device flash, which will result in overriding
+//! the older EEPROM settings. This essentially freezes the EEPROM-based settings in time from the
+//! switch over.
+//!
+//! Settings are stored in flash because of the restrictive size of EEPROM on the device making it
+//! impossible to save domain names for a named broker into EEPROM, as the storage is only 128
```
Comment (jordens) [inline src/settings/global_settings.rs:17]:
```suggestion
//! impossible to save domain names for a named broker into EEPROM, as the available board data storage is only 64
```

**Review by jordens — CHANGES_REQUESTED** [review]
LGTM with these slight changes.

**Review by jordens — APPROVED** [review]

**Pull request merged**